spec/tictactoe_spec.rb

```diff
@@ -1,11 +1,46 @@
-require 'spec_helper'
-require 'tictactoe'
+require "spec_helper"
+require "tictactoe"
+require "presenter"
+require "fake_standard_out"
 
 RSpec.describe "TicTacToe" do
+  class FakePresenter
+    def display_board
+      <<~BOARD
+          |  |
+        --+--+--
+          |  |
+        --+--+--
+          |  |
+      BOARD
+    end
+  end
+  class FakeDisplay
+    def initialize(stdout)
+      @stdout = stdout
+    end
+
+    def output(message)
+      @stdout.puts message
+    end
+  end
   describe "#run" do
-    it "returns true" do
-      tictactoe = TicTacToe.new
-      expect(tictactoe.run).to eql true
+    it "prints an empty 3 x 3 Tic-Tac-Toe board" do
+      stdout = FakeStandardOut.new
+      display = FakeDisplay.new(stdout)
+      presenter = FakePresenter.new
+      tictactoe = TicTacToe.new(presenter, display)
```

Comment:
It looks like we have to do a lot of setup to instantiate this object, but the object only has two immediate dependencies.  The objects that are constructed are also at varying levels of abstraction. That suggests to me that this tests know more details that it needs to. What could we do to make this test simpler?

Reply:
I removed the fakes, except FakeStandardOut so it's more of an integration test, but not sure what else to do to simplify?

Reply:
If we wanted to unit test at this level, we could create test doubles that are customized for `TicTacToe`. For instance, we could create a `TestDisplay` that provides that provides an `#output` method, and we could do something with that, but that test double wouldn't need to accept a `stdout`. Accepting a `stdout` is a detail of the real `Display`, and that detail is not pertintent to `TicTacToe`.  Is that helpful?

Reply:
Yes